packages/grid/x-data-grid/src/internals/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -166,12 +168,28 @@ export const useGridKeyboardNavigation = (
   );
 
   const handleColumnHeaderKeyDown = React.useCallback<
-    GridEventListener<GridEvents.columnHeaderNavigationKeyDown>
+    GridEventListener<GridEvents.columnHeaderKeyDown>
   >(
     (params, event) => {
-      if (!params.field) {
+      const headerTitleNode = event.currentTarget.querySelector(
+        `.${gridClasses.columnHeaderTitleContainerContent}`,
+      );
+      const isFromCustomHeader =
```

Comment:
To keep consistency. We're only checking if the target is inside the content, not if `renderCell` was used.

```suggestion
      const isFromInsideContent =
```